docs/CODE_GUIDELINES.md

```diff
@@ -341,11 +353,12 @@ int const * const someIntPointer;
 
 ### 5.4. Initialization
 
-Make sure that variables are initialized appropriately at declaration or soon afterwards. This is especially important for fundamental type variables that do not have any constructor. Zero-initialize with `{}`.
+Make sure that variables are initialized appropriately at declaration or soon afterwards. This is especially important for fundamental type variables that do not have any constructor to zero-initialize with `{}` (e.g. structs, pointers).
+This syntax is less ambiguous and disallows narrowing (cf. [ISO C++ guidelines](http://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Res-list)).
 
 ✅ Good:
 ```cpp
-int x{};
+int x{0};
 int* y{};
```

Comment:
```c++
int* y{0};
```

A comment in another thread got me thinking, for primitive types, we should always include a value in the initializer list, because it offers explicit type checking for the value. For anything beyond a primitive type `{}` is fine.

Reply:
shouldnt it be `int* y{nullptr};` then?

Reply:
for pointers specify `nullptr` not change the result in any case is nullptr
you could says that it is preferible or more readable

Reply:
@garbear specified line its a pointer, anyway i will clarify the point

Reply:
Thanks, been staring at a screen for too long. Updated my comment. I'll have to come back tomorrow to reflect on what's been said here.

Reply:
i tried make more clear point `5.4. Initialization` let me know